course5/homework04/homework04.cc

```diff
@@ -0,0 +1,122 @@
+#include <iostream>
+#include <cstdio>
+#include <cstdlib>
+#include <stdexcept>
+#include <string>
+
+/***
+Define a “Stream” class, that allows for easy printing of values to a file
+- Let the class manage the file: ie open the file in constructor and close it in destructor.
+- Overload operator<< for the class, so that it can be used to print some types.
+- Define a global variable of the “Stream” type, call it, say, “Cout”, that will write to stdout.
+***/
+
+class Stream
+{
+public:
+	// Constructors
+	Stream() // default
+	{
+		throw std::runtime_error("Stream must be constructed with a filename: Stream('filename.txt')");
+	}
+	Stream(const Stream&) // copy
+	{
+		throw std::runtime_error("Stream object cannot be copied");
+	}
+	Stream(Stream&& src) // move
+	{
+		fname = src.fname;
+		fp = src.fp;
```

Comment:
```src``` will cease to exists immediately after the function is over, but its destructor will be called, so fp will be closed.
You need set src.fp to some value that will stop src's destructor from closing it.

This exposes a gap in your testing: once you implemented the move semantic, you should test it. Write a code that uses it.